classification.yaml

```diff
@@ -0,0 +1,791 @@
+    - Run QG generation against any major repo (e.g. `/repo/main`).
+DMS-MWBT-REQ-0084:
+  status: feature-complete
+  priority: 1a
+  script:
+    - Run QG generation for the DRP pipeline against any major repo (e.g. `/repo/main`).
+  reuse_script: DMS-MWBT-REQ-0083
+DMS-MWBT-REQ-0086:
+  status: feature-incomplete
+  priority: 1a
+  script:
+    - Run QG generation in an RSP notebook.
+  notes:
+    Needs a pipetask Python API equivalent.
+    Could use SimplePipelineExecutor to satisfy this, but it'd be better to use something we feel comfortable advertising to notebook users, rather than a placeholder.
+DMS-MWBT-REQ-0085:
+  status: feature-complete
+  priority: 1a
+  script:
+    - Run QG generation against any major repo (e.g. `/repo/main`).
+  reuse_script: DMS-MWBT-REQ-0083
+DMS-MWBT-REQ-0025:
+  status: feature-complete
+  priority: 1a
+  script:
+    - Make an empty repo with the default configuration.
+    - Make an empty repo with configuration that overrides a formatter.
+    - Make an empty repo with configuration that changes a StorageClass's Python type.
+    - Put and get the same datasets to all repos.
+DMS-MWBT-REQ-0026:
+  status: hackable
+  priority: 1a
+  script:
+    - Add a butler method that prints/saves the full expanded config (it's just a private attribute right now).
+    - Use it.
+DMS-MWBT-REQ-0027:
+  status: feature-incomplete
+  priority: 1a
+  script:
+    - Run pipeline with repo config override that uses InMemoryDatastore.
+  missing:
+    - Need to resolve issues with InMemoryDatastore, or at least those inside the middleware (we could live with the "use at your own risk" aliasing issue, as that's probably impossible to really fix).
+DMS-MWBT-REQ-0030:
+  status: feature-complete
+  priority: 1a
+  script:
+    - Make an empty repo with a POSIX datastore, do get/put.
+DMS-MWBT-REQ-0031:
+  status: feature-complete
+  priority: 1a
+  script:
+    - Make an empty repo with an S3 datastore, do get/put.
+DMS-MWBT-REQ-0028:
+  status: feature-incomplete
+  priority: 1a
+  script:
+    - Make an empty repo with a VOSpace datastore, do get/put.
+  missing:
+    - A VOSpace datastore implementation.
+  notes:
+    Is this actually distinct from S3?
+DMS-MWBT-REQ-0029:
+  status: feature-incomplete
+  priority: 1a
+  script:
+    - Make an empty repo with a VOSpace datastore, do get/put.
+  reuse_script: DMS-MWBT-REQ-0028
+  missing:
+    - A VOSpace datastore implementation.
+  notes:
+    Is this actually distinct from S3?
+DMS-MWBT-REQ-0032:
+  status: unclear
+  priority: 1a
+  notes:
+    Looking at the motivating use case, this requirement seems to really be about supporting both HDF5 and FITS.  But the spec seems to suggest we need to be able to support "any" scientific data format, which is not really enumerable.
+    It also seems to presume that we would have some classes that can be serialized identically to different formats.  But we don't, and I don't really see us adding any just to satisfy this test.
```

Comment:
I just didn't think this was a verifiable requirement, but @timj wrote it (I think) and thinks it means something trivial.  Maybe it just needs to be reworded?

Reply:
You are over-interpreting this one. It's meant to be the simplest possible requirement that any science data that we `butler.put` we must be able to retrieve via `butler.get`. It's one of the reasons why we store the formatter in the datastore so that we know which datastore was used to write it.  It was setting a boundary condition for ARCH2. ARCH2 is broader in that it's requiring that we can configure different formatters for the same dataset type (which we can do trivially with a chained datastore). We can already do a run with one mapping for dataset type to formatter and then do a second processing run with a different formatter. They will be able to be read back even if the config changes later.

Reply:
If @timj 's interpretation is the one we go with then we can claim this easily now.

Reply:
By the design of the config system and with chained datastores  (or a local override of a formatter configuration) we can show that you can write different file formats if you want for the same python object and we will always read it back using the correct formatter class.